### PR TITLE
Attesting and block proposing logic

### LINKED ISSUE #203: [Validator] hook in attesting / block proposing logic
https://github.com/ethereum/eth2.0-specs/blob/dev/specs/validator/0_beacon-chain-validator.md#beacon-chain-responsibilities

### PR DESCRIPTION
**Pull request by mpetrunic**
- rpc calls for block producing/publishing
- rpc calls for attestation producing/publishing
- object signing

resolves #203

### CODE REVIEW

File: src/validator/services/attestation.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      aggregationBitfield: Buffer.alloc(committeeLength, 0)
+    };
+    await this.storeAttestation(attestation);
+    await this.rpcClient.validator.publishAttestation(attestation);
+    logger.info(`[Validator] Signed and publish new attestation`);
+    return attestation;
+  }
+
+  private async isConflictingAttestation(other: AttestationData): Promise<boolean> {
+    // TODO: Fetch last signed attestation data and check
```
Comment (GregTheGreek) [inline src/validator/services/attestation.ts:55]: next step is store published attestations to disk and pull them from there

**Review by GregTheGreek — DISMISSED** [review]

File: src/validator/services/attestation.ts (hunk 1 above)
Comment (mpetrunic) [reply src/validator/services/attestation.ts:55]: yeah it was separate item on the list and its probably easier to review in this small increments.

File: src/validator/services/attestation.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    const signature = this.privateKey.signMessage(
+      hashTreeRoot(attestationDataAndCustodyBit, AttestationDataAndCustodyBit),
+      getDomainFromFork(
+        fork,
+        slotToEpoch(slot),
+        Domain.ATTESTATION
+      )
+    ).toBytesCompressed();
+    //TODO: get real committee length
+    const committeeLength = 8;
```
Comment (mpetrunic) [inline src/validator/services/attestation.ts]: @GregTheGreek @wemeetagain What should we do about this? Fetch beacon state and calculate:
https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#get_epoch_committee_count ?

Comment (wemeetagain) [reply src/validator/services/attestation.ts]: Should get it from a `CommitteeAssignment` object from `getCommitteeAssignment`

Comment (mpetrunic) [reply src/validator/services/attestation.ts]: ahh nice, not sure how I forgot about that

**Review by GregTheGreek — APPROVED** [review]

**Review by wemeetagain — APPROVED** [review]

**Pull request merged**